backend/compact-connect/tests/unit/test_app.py

```diff
@@ -15,6 +15,28 @@
 
 class TestApp(TestCase):
 
+    def _when_testing_production_stack_synth(self):
```

Comment:
I like these simple abstractions to clean up the test code. Can we give the `_when...` helper methods some more specific names to make it clear where they're used? These, for example, are returning context, while the one in `test_api.py` is actually synthesizing the app.